internal/controller/instaslice_daemonset_test.go

```diff
@@ -92,28 +67,50 @@ func TestCleanUp(t *testing.T) {
 			},
 		},
 	}
-	fakeClient.Create(context.Background(), instaslice)
+	errCreatingInstaSlice := fakeClient.Create(context.Background(), instaslice)
+	assert.NoError(t, errCreatingInstaSlice)
+
+	// Create a fake Node resource
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "node-1",
+		},
+		Status: v1.NodeStatus{
+			//TODO: add test cases for other resources
+			Capacity: v1.ResourceList{
+				"org.instaslice/uid-1": resource.MustParse("1"),
+			},
+		},
+	}
+	errCreatingNode := fakeClient.Create(context.Background(), node)
+	assert.NoError(t, errCreatingNode)
 
 	// Set the NODE_NAME environment variable
 	os.Setenv("NODE_NAME", "node-1")
 	defer os.Unsetenv("NODE_NAME")
 
+	// Keeping it around as it may be needed later
 	// Create a fake Pod resource
-	pod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			UID:       "pod-uid-1",
-			Name:      "pod-name-1",
-			Namespace: "default",
-		},
-	}
+	// pod := &v1.Pod{
```

Comment:
would it make sense to purge these lines if they aren't required anymore?

Reply:
I kept this on purpose; I can imagine a fake pod object needed to write other unit tests.

Reply:
Oh ok